core/src/main/java/com/hevelian/identity/core/specification/EntitySpecification.java

```diff
@@ -0,0 +1,87 @@
+package com.hevelian.identity.core.specification;
+
+import com.hevelian.identity.core.exc.NotImplementedException;
+import com.hevelian.identity.core.model.AbstractEntity;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * Creates {@link Predicate} objects for Data Base query.
+ *
+ * @author yshymkiv
+ */
+public class EntitySpecification<T extends AbstractEntity> implements Specification<T> {
+  private final static String FROM_DATE = "From";
+  private final static int LENGTH_STR_FROM = 4;
+  private final static String TO_DATE = "To";
+  private final static int LENGTH_STR_TO = 2;
+
+  private SearchCriteria criteria;
+
+  public EntitySpecification(SearchCriteria param) {
+    this.criteria = param;
+  }
+
+  @Override
+  public Predicate toPredicate(Root<T> root, CriteriaQuery<?> query, CriteriaBuilder builder) {
+    String key = criteria.getKey();
+    Object value = criteria.getValue();
+
+    if (value == null) {
+      return null;
+    }
+
+    Predicate predicate = null;
+    if (root.getModel().getAttributes().stream().anyMatch(a -> Objects.equals(a.getName(), key))) {
```

Comment:
It's imposible because when comes filtering date parameter (dateCreatedFrom or dateCreatedTo) we gets exception. This fields don't exist in Entity classes.

Reply:
That's weird because Objects.equals(a.getName(), key)  looks like it checks exactly by name. Let's discuss this separately.